internal/provider/data_source_http.go

```diff
@@ -136,7 +151,33 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 		method = "GET"
 	}
 
-	client := &http.Client{}
+	caCertificate := model.CaCertificate
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
+
+	if !model.Insecure.IsNull() {
+		tr.TLSClientConfig.InsecureSkipVerify = model.Insecure.Value
+	}
+
+	// Use `ca_cert_pem` cert pool
+	if !caCertificate.IsNull() {
+		caCertPool := x509.NewCertPool()
+		if ok := caCertPool.AppendCertsFromPEM([]byte(caCertificate.Value)); !ok {
+			resp.Diagnostics.AddError(
+				"Error tls",
```

Comment:
Can you make this error message more descriptive? Maybe something like "Error configuring TLS client"

Reply:
added.